app/registry.hs

```diff
@@ -380,7 +380,7 @@ validatePublicKey :: MonadFail m => Text -> m Owner
 validatePublicKey text =
   if      not ((Char.toLower <$> s) == s)
   then fail $ "Public key not all-lowercase: " <> s
-  else Owner <$> decodeBech32 "Owner Public key" Bech32.decode text
+  else Owner <$> decodeBech32 "Owner Public key" publicKeyHRP text
```

Comment:
So we switch from strict to lenient PKs here.

cc @KtorZ

Reply:
Yes! The bech32 spec _recommend_ using bech32 for strings up to a certain length (to get the most of the guarantees in the error detection). However in Cardano, most of our cases are beyond that recommended length. In the `bech32` library, `decodeLenient` means that it won't check that the length of the decoded string is below that limit, but it'll still perform all other validations of the encoded string!